### ISSUE TITLE
RenderMapping is not applied to DSL.schema(Name)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#9386]
The following program:

```java
public class X {
    public static void main(String[] args) {
        DSLContext ctx = DSL.using(SQLDialect.H2,
            new Settings().withRenderMapping(
                new RenderMapping().withSchemata(
                    new MappedSchema().withInput("A").withOutput("C"),
                    new MappedSchema().withInput("B").withOutput("D")
                )));

        System.out.println(ctx.createSchema(name("A")));
        System.out.println(ctx.alterSchema(name("A")).renameTo("B"));
        System.out.println(ctx.dropSchema(name("A")));
        System.out.println(ctx.createTable(name("A", "T")).column("C", INTEGER));
    }
}
```

... produces:

```sql
create schema "A"
alter schema "A" rename to "B"
drop schema "A"
create table "C"."T"(
  "C" int null
)
```

When it should produce

```sql
create schema "C"
alter schema "C" rename to "D"
drop schema "C"
create table "C"."T"(
  "C" int null
)
```

**Comment by lukaseder** [comment]
Hmm, tables map themselves after mapping the schema. There are probably more subtle bugs hiding in this implementation, for example:

- Transitive mappings might apply when they intuitively shouldn't: `A -> B -> C`
- When a table maps itself, it maps itself based on the original schema, not the mapped schema

**Comment by lukaseder** [comment]
Fixed in 3.13.0. Backport scheduled for 3.12.2 (#9387)
